File: csharp/ToolGood.Algorithm/Internals/Functions/DateTimes/Function_DAYS.cs
using System;
using System.Text;
using ToolGood.Algorithm.Enums;

namespace ToolGood.Algorithm.Internals.Functions.DateTimes
{
	internal sealed class Function_DAYS : Function_2
	{
		public Function_DAYS(FunctionBase func1, FunctionBase func2) : base(func1, func2)
		{
		}

		public Function_DAYS(FunctionBase[] funcs) : base(funcs)
		{
		}

		public override string Name => "Days";

		public override Operand Evaluate(AlgorithmEngine engine, Func<AlgorithmEngine, string, Operand> tempParameter)
		{
			var args1 = GetDate_1(engine, tempParameter);
			if (args1.IsError) { return args1; }

			var args2 = GetDate_2(engine, tempParameter);
			if (args2.IsError) { return args2; }

			var endDate = args1.DateValue.ToDateTime();
			var startDate = args2.DateValue.ToDateTime();

			return Operand.Create((endDate.Date - startDate.Date).Days);
		}
		public override OperandType GetResultType()
		{
			return OperandType.NUMBER;
		}
	}
}

File: java/toolgood.algorithm/src/main/java/toolgood/algorithm/internals/functions/dateTimes/Function_DAYS.java
package toolgood.algorithm.internals.functions.dateTimes;

import java.util.List;
import org.joda.time.DateTime;
import org.joda.time.DateTimeZone;
import org.joda.time.Days;
import toolgood.algorithm.AlgorithmEngine;
import toolgood.algorithm.Operand;
import toolgood.algorithm.enums.OperandType;
import toolgood.algorithm.internals.NoneEngine;
import toolgood.algorithm.internals.ParameterType;
import toolgood.algorithm.internals.functions.FunctionBase;
import toolgood.algorithm.internals.functions.Function_2;

final class Function_DAYS extends Function_2 {
    public Function_DAYS(FunctionBase func1, FunctionBase func2) {
        super(func1, func2);
    }

    public Function_DAYS(FunctionBase[] funcs) {
        super(funcs);
        if (funcs.length != 2) {
            throw new IllegalArgumentException("Function '" + Name() + "' requires exactly 2 parameters.");
        }
    }

    @Override
    public String Name() {
        return "Days";
    }

    @Override
    public Operand Evaluate(AlgorithmEngine engine, java.util.function.BiFunction<AlgorithmEngine, String, Operand> tempParameter) {
        Operand args1 = GetDate_1(engine, tempParameter);
        if (args1.IsErrorOrNone()) { return args1; }

        Operand args2 = GetDate_2(engine, tempParameter);
        if (args2.IsErrorOrNone()) { return args2; }

        DateTime endDate = args1.DateValue().ToDateTime();
        DateTime startDate = args2.DateValue().ToDateTime();

        return Operand.Create(Days.daysBetween(startDate.withTimeAtStartOfDay(), endDate.withTimeAtStartOfDay()).getDays());
    }

    @Override
    public OperandType GetResultType() {
        return OperandType.NUMBER;
    }

    @Override
    void GetParameterTypes(NoneEngine noneEngine, List<ParameterType> result, OperandType operandType, String op, String val) {
        func1.GetParameterTypes(noneEngine, result, OperandType.DATE);
        func2.GetParameterTypes(noneEngine, result, OperandType.DATE);
    }
}
